samples/trace-utility-network/src/main/java/com/esri/arcgismaps/sample/traceutilitynetwork/components/TraceUtilityNetworkViewModel.kt

```diff
@@ -0,0 +1,621 @@
+import com.arcgismaps.mapping.view.Graphic
+import com.arcgismaps.mapping.view.GraphicsOverlay
+import com.arcgismaps.mapping.view.IdentifyLayerResult
+import com.arcgismaps.mapping.view.ScreenCoordinate
+import com.arcgismaps.portal.Portal
+import com.arcgismaps.toolkit.geoviewcompose.MapViewProxy
+import com.arcgismaps.utilitynetworks.UtilityDomainNetwork
+import com.arcgismaps.utilitynetworks.UtilityElement
+import com.arcgismaps.utilitynetworks.UtilityElementTraceResult
+import com.arcgismaps.utilitynetworks.UtilityNetwork
+import com.arcgismaps.utilitynetworks.UtilityNetworkSource
+import com.arcgismaps.utilitynetworks.UtilityNetworkSourceType
+import com.arcgismaps.utilitynetworks.UtilityTerminal
+import com.arcgismaps.utilitynetworks.UtilityTier
+import com.arcgismaps.utilitynetworks.UtilityTraceParameters
+import com.arcgismaps.utilitynetworks.UtilityTraceType
+import com.esri.arcgismaps.sample.sampleslib.components.MessageDialogViewModel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.launch
+
+class TraceUtilityNetworkViewModel(application: Application) : AndroidViewModel(application) {
+
+    // The textual hint shown to the user
+    private val _hint = MutableStateFlow<String?>(null)
+    val hint = _hint.asStateFlow()
+
+    // Is trace utility network enabled
+    private val _canTrace = MutableStateFlow(false)
+    val canTrace = _canTrace.asStateFlow()
+
+    // The trace state used for the sample
+    private val _traceState = MutableStateFlow(TraceState.NOT_STARTED)
+    val traceState = _traceState.asStateFlow()
+
+    // Currently selected utility trace type
+    private val _selectedTraceType = MutableStateFlow<UtilityTraceType?>(null)
+    val selectedTraceType = _selectedTraceType.asStateFlow()
+
+    // Currently selected point type (start/barrier)
+    private val _selectedPointType = MutableStateFlow(PointType.None)
+    val selectedPointType = _selectedPointType.asStateFlow()
+
+    // Terminal configuration options (high/low)
+    private val _terminalConfigurationOptions = MutableStateFlow<List<UtilityTerminal>>(listOf())
+    val terminalConfigurationOptions = _terminalConfigurationOptions.asStateFlow()
+
+    // Currently selected terminal configuration
+    private var _selectedTerminalConfigurationIndex = MutableStateFlow<Int?>(null)
+
+    // ArcGISMap holding the UtilityNetwork and operational layers
+    val arcGISMap = ArcGISMap(
+        item = PortalItem(
+            portal = Portal.arcGISOnline(connection = Portal.Connection.Authenticated),
+            itemId = NAPERVILLE_ELECTRICAL_NETWORK_ITEM_ID
+        )
+    ).apply {
+        // Add the map with streets night vector basemap
+        setBasemap(Basemap(BasemapStyle.ArcGISStreetsNight))
+    }
+
+    // Used to handle map view animations
+    val mapViewProxy = MapViewProxy()
+
+    // The utility network used for tracing.
+    private val utilityNetwork: UtilityNetwork
```

Comment:
What's the advantage of setting a get() instead of assigning the value with an = ?